src/app/3d/qgs3dmapcanvaswidget.h

```diff
@@ -51,6 +51,11 @@ class APP_EXPORT Qgs3DMapCanvasDockWidget : public QgsDockWidget
 
     Qgs3DMapToolMeasureLine *measurementLineTool() { return  mMapToolMeasureLine; }
 
+    void setDocked( bool docked );
+
+  signals:
+    void toggleDockMode( bool docked );
```

Comment:
Can we rename this signal to e.g. `toggleDockModeRequested` ? It should be hopefully more self-explanatory that it only asks someone else to change the dock mode of the widget rather than doing it by itself and informing others about the change of state...